File: backend/src/main/kotlin/com/example/backend/dto/PostDto.java
package com.example.backend.dto;

import lombok.Data;
import java.time.LocalDateTime;

public class PostDto {
    
    @Data
    public static class Request {
        private String title;
        private String content;
        private String author;
    }
    
    @Data
    public static class Response {
        private Long id;
        private String title;
        private String content;
        private String author;
        private int likesCount;
        private int commentsCount;
        private LocalDateTime createdAt;
        private LocalDateTime updatedAt;
    }
    
    @Data
    public static class ListResponse {
        private Long id;
        private String title;
        private String content;
        private String author;
        private int likesCount;
        private int commentsCount;
        private LocalDateTime createdAt;
    }
} 

File: backend/src/main/kotlin/com/example/backend/dto/PostDto.kt
package com.example.backend.dto

import java.time.LocalDateTime

object PostDto {
    data class Request(
        var title: String,
        var content: String,
        var author: String
    )

    data class Response(
        var id: Long,
        var title: String,
        var content: String,
        var author: String,
        var likesCount: Int,
        var commentsCount: Int,
        var createdAt: LocalDateTime,
        var updatedAt: LocalDateTime
    )

    data class ListResponse(
        var id: Long,
        var title: String,
        var content: String,
        var author: String,
        var likesCount: Int,
        var commentsCount: Int,
        var createdAt: LocalDateTime
    )
} 